server/index.js

```diff
@@ -68,6 +68,86 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  const sql = `
+    select *
+    from "carts"`;
+  db.query(sql)
+    .then(result => {
+      const carts = result.rows;
+      res.status(200).json(carts);
+    })
+    .catch(err => next(err));
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = parseInt(req.body.productId);
+  if (!productId) {
+    res.status(400).json({
+      error: 'Please enter a productId number'
+    });
+    return;
+  }
+  const sqlPrice = `
+  select "price"
+    from "products"
+   where "productId" = $1`;
+  const value = [productId];
+  db.query(sqlPrice, value)
+    .then(priceResult => {
+      if (!priceResult.rows.length) {
+        res.status(400).json({ error: 'No product with this ProductId exists' }
```

Comment:
Here, instead of responding, you should `throw` to shortcut to the `catch()` callback of your promise chain.

```js
throw new ClientError('No product with this productId exists', 400)
```